src/devices/Tlc59711/Tlc59711.cs

```diff
@@ -0,0 +1,157 @@
+using System.Text;
+using System.Text.RegularExpressions;
+using Iot.Device.Spi;
+
+namespace Iot.Device.Tlc59711
+{
+    /// <summary>
+    /// On-board LED on the device.
+    /// </summary>
+    public class Tlc59711 : IDisposable
+    {
+        private const int ChannelsPerDriver = 12;
+
+        private readonly ushort[] _pwmBuffer;
+        private readonly int _numberOfDrivers;
+        private SpiDevice _spiDevice;
+        private byte _brightnessRed;
+        private byte _brightnessGreen;
+        private byte _brightnessBlue;
+        private byte[] _dataToWrite;
+
+        /// <summary>
+        /// Creates a new instance of the Tlc59711.
+        /// </summary>
+        public Tlc59711(SpiDevice spiDevice, int numberOfDrivers = 1)
+        {
+            _numberOfDrivers = numberOfDrivers;
+
+            _spiDevice = spiDevice ?? throw new ArgumentNullException(nameof(spiDevice));
+
+            // default 100% brigthness
+            _brightnessRed = _brightnessGreen = _brightnessBlue = 0x7f;
+
+            // 12 channels per driver
+            _pwmBuffer = new ushort[numberOfDrivers * ChannelsPerDriver];
+
+            // Two bytes per channel, plus 1 byte for the command, and 3 bytes for the global brightness control
+            _dataToWrite = new byte[_numberOfDrivers * (ChannelsPerDriver * 2 + 4)];
+        }
+
+        /// <summary>
+        /// Disposes SpiDevice instances
+        /// </summary>
+        public void Dispose()
+        {
+            _spiDevice?.Dispose();
+            _spiDevice = null;
+        }
+
+        /// <summary>
+        /// Call this to set the PWM level for a channel.
+        /// </summary>
+        /// <param name="channel">Channel, 0-based</param>
+        /// <param name="pwm">PWM level (0 = minimum, 65535 = maximum)</param>
+        public void SetPWM(int channel, ushort pwm)
+        {
+            if (channel > ChannelsPerDriver * _numberOfDrivers)
+            {
+                throw new ArgumentOutOfRangeException("channel");
+            }
+
+            _pwmBuffer[channel] = pwm;
+        }
+
+        /// <summary>
+        /// Call this to set the RGB value for a group of 3 channels
+        /// </summary>
+        /// <param name="ledNumber">LED number (channel number of the "red" pin divided by 3)</param>
+        /// <param name="red">Red level (0 = minimum, 65535 = maximum)</param>
+        /// <param name="green">Green level (0 = minimum, 65535 = maximum)</param>
+        /// <param name="blue">Blue level (0 = minimum, 65535 = maximum)</param>
+        public void SetLED(int ledNumber, ushort red, ushort green, ushort blue)
+        {
+            SetPWM((byte)(ledNumber * 3), red);
+            SetPWM((byte)(ledNumber * 3 + 1), green);
+            SetPWM((byte)(ledNumber * 3 + 2), blue);
+        }
+
+        /// <summary>
+        /// Call this after every change to write the new PWM levels to the device.
+        /// </summary>
+        public void Write()
```

Comment:
this should be called automatically, user should not have to do this, you should rather provide API for changing all colors at the same time (i.e. with Span or image buffer - i.e. similarly as Wsb2812b code does it)